src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
 
+        if (table[hash] == null) {
+            table[hash] = newNode;
+        } else {
+            Node<K, V> current = table[hash];
+            while (current != null) {
+                if (current.hash == hash
+                        && (current.key == key || (key != null && key.equals(current.key)))) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = newNode;
+                    break;
+                }
+                current = current.next;
+            }
+        }
+
+        size++;
+        resizeIfNeeded();
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        Node<K, V> node = table[hash];
+
+        while (node != null) {
+            if (node.hash == hash
+                    && (node.key == key || (key != null && key.equals(node.key)))) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % DEFAULT_CAPACITY);
+    }
+
+    private void resizeIfNeeded() {
+        if (size > threshold) {
+            resize();
+        }
+    }
+
+    private void resize() {
+        int newCapacity = table.length * 2;
+        threshold = (int) (newCapacity * LOAD_FACTOR);
+        Node<K, V>[] newTable = new Node[newCapacity];
+
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                int newHash = Math.abs(node.hash % newCapacity);
+                Node<K, V> next = node.next;
+                node.next = newTable[newHash];
+                newTable[newHash] = node;
+                node = next;
+            }
+        }
+        table = newTable;
+    }
```

Comment:
Performance issue: The resize method can be optimized. Currently, you are iterating through all nodes in the old table and for each node, you are recalculating the hash and inserting it into the new table. This can be quite inefficient for large tables. A better approach would be to only rehash the keys that need to be moved to a different index in the new table. You can do this by checking if the bit at the position of the new size in the hash is set. If it is, the node needs to be moved to `index + oldCapacity` in the new table.